### PR TITLE
Miscellaneous CMake improvements

### PR DESCRIPTION
**Pull request by henryleberre**
This pull request contains the changes that were required for Silo to compile and link on my system using CMake. `libm` and `libdl` had to be given to the linker, and a destination library directory name (`lib`) had to be specified in Silo's `install` command.

### CODE REVIEW

**Comment by henryleberre** [comment]
2nd commit: If a user manually specified their own `CMAKE_INSTALL_PREFIX`, Silo would still set its "default" prefix path. I fixed it by adding `if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)`.

**Comment by henryleberre** [comment]
@markcmiller86 Thank you for your quick reply. With these small tweaks, I was able to get Silo integrated into our CMake build system. I don't use a mac, but I just read that libdl is not a thing in that ecosystem. CMake conveniently provides `${CMAKE_DL_LIBS}` as a cross-platform replacement. I will make that change, and combine my commits by rebasing so that they don't clutter the repository's history. It should then be ready for review.

**Comment by markcmiller86** [comment]
@biagas when you have some time can you take a look?

File: CMakeLists.txt [hunk 1]
```diff
@@ -646,11 +652,14 @@ endif()
...
 # silock
 ##
-if(NOT WIN32)
+if(SILO_ENABLE_SILOCK AND NOT WIN32)
     add_executable(silock
         ${Silo_SOURCE_DIR}/tools/silock/silock.c)
     add_dependencies(silock silo)
     target_link_libraries(silock $<TARGET_LINKER_FILE:silo>)
+    if(UNIX)
+        target_link_libraries(silock m dl)
```
Comment (biagas) [inline CMakeLists.txt:661]:
```suggestion
        target_link_libraries(silock m ${CMAKE_DL_LIBS})
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -570,14 +572,17 @@ if(SILO_ENABLE_HDF5 AND HDF5_FOUND)
...
 add_library(silo ${silo_library_sources})
-target_compile_definitions(silo PRIVATE ${SILO_COMPILE_DEFINES})
-add_dependencies(silo pdb_detect)
-target_include_directories(silo PRIVATE ${silo_library_include_dirs})
 
 if(UNIX)
     target_link_libraries(silo m)
 endif()
 
+target_link_libraries(silo "${CMAKE_DL_LIBS}")
```
Comment (biagas) [inline CMakeLists.txt:580]:
```suggestion
target_link_libraries(silo ${CMAKE_DL_LIBS})
```

Comment (biagas) [reply CMakeLists.txt:580]: Just wondering if ${CMAKE_DL_LIBS} should simply be added to line 577?

File: tools/browser/CMakeLists.txt [hunk 3]
```diff
@@ -95,6 +95,8 @@ if(UNIX)
     target_link_libraries(browser m)
 endif()
 
+target_link_libraries(browser "${CMAKE_DL_LIBS}")
```
Comment (biagas) [inline tools/browser/CMakeLists.txt:98]:
```suggestion
target_link_libraries(browser ${CMAKE_DL_LIBS})
```

**Review by biagas — CHANGES_REQUESTED** [review]

**Comment by markcmiller86** [comment]
thanks so much @henryleberre 🎉

**Pull request merged**